pkg/crc/segment/segment.go

```diff
@@ -0,0 +1,74 @@
+package segment
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/pborman/uuid"
+	"github.com/segmentio/analytics-go"
+)
+
+func NewSegmentClient() (analytics.Client, error) {
+	return analytics.NewWithConfig("cvpHsNcmGCJqVzf6YxrSnVlwFSAZaYtp", analytics.Config{
+		DefaultContext: &analytics.Context{
+			App: analytics.AppInfo{
+				Name:    constants.DefaultName,
+				Version: version.GetCRCVersion(),
+			},
+		},
+	})
+}
+
+func UploadToSegment(client analytics.Client, err error) error {
```

Comment:
Wrap segment library and use the config to decide wether or not send to segment:

```
type Client struct {
  segmentClient analytics.Client
  config Config // crc configuration object
}

func (c *Client) Upload() {
  if config.Get(key).AsBool() {
    // send to segment
  }
}

```

It will then be easy to add a unit test that verify no data are sent to segment if the config key is set.
A unit test that verifies what is sent to Segment is also a good thing to have.